src/main/java/frc/robot/commands/MoveToGamepiece.java

```diff
@@ -9,44 +9,46 @@
 import frc.robot.subsystems.Vision;
 
 public class MoveToGamepiece extends Command {
-    private SwerveDrivetrain drivetrain;
-    private Vision vision;
+    private final SwerveDrivetrain drivetrain;
+    private final Vision vision;
     private double startTime;
     private final PIDController turningPIDController;
     private final PIDController movingPIDController;
-    private ChassisSpeeds driveStates;
-    private double ychange;
-    private double xchange;
+    private double yChange;
+    private double pieceNotSeenCounter = 0;
 
 
 
     public MoveToGamepiece(SwerveDrivetrain drivetrain, Vision vision) {
         this.drivetrain = drivetrain;
         this.vision = vision;
-        addRequirements(drivetrain);
         turningPIDController = new PIDController(Constants.MOVE_TO_GAMEPIECE_TURNING_P, 0, Constants.MOVE_TO_GAMEPIECE_TURNING_D);
         movingPIDController = new PIDController(Constants.MOVE_TO_GAMEPIECE_MOVING_P, 0, 0);
-
+        addRequirements(drivetrain);
     }
 
     @Override
     public void initialize() {
         startTime = Timer.getFPGATimestamp();
+        pieceNotSeenCounter = 0;
     }
 
     @Override
     public void execute() {
-        ychange = vision.getPieceOffestAngleY() - Constants.LIMELIGHT_MOVE_TO_PIECE_DESIRED_Y;
-        xchange = vision.getPieceOffestAngleX() - Constants.LIMELIGHT_MOVE_TO_PIECE_DESIRED_X;
-        if (vision.isPieceSeen() && (ychange > Constants.MOVE_TO_GAMEPIECE_THRESHOLD)) {
-        driveStates = new ChassisSpeeds(movingPIDController.calculate(ychange), 0, turningPIDController.calculate(xchange));
-        drivetrain.drive(driveStates);
+        yChange = vision.getPieceOffestAngleY() - Constants.LIMELIGHT_MOVE_TO_PIECE_DESIRED_Y;
+        double xChange = vision.getPieceOffestAngleX() - Constants.LIMELIGHT_MOVE_TO_PIECE_DESIRED_X;
+        if (yChange == 0){
```

Comment:
That does not look right - that should use the `tv`
I think the logic for `isPieceSeen` should have been copied over.

Reply:
I think it is better now